torch/_inductor/fx_passes/misc_patterns.py

```diff
@@ -69,6 +69,52 @@ def randperm_index_replacement(x, slice_shape):
         scalar_workaround={"slice_shape": 42},
     )
 
+    # Float8 training patterns
```

Comment:
I can also move the pattern to torchao/float8 folder if we think that's a better place.

Reply:
Yea, i think since this is very float8 specific, let's put it there